test/mitmproxy/test_cmdline.py

```diff
@@ -53,6 +54,12 @@ def test_parse_server_spec():
         "http://")
 
 
+def test_parse_upstream_auth():
+    tutils.raises("Invalid upstream auth specification", cmdline.parse_upstream_auth, "")
+    assert cmdline.parse_upstream_auth(
+        "test:test") == "Basic" + " " + base64.b64encode("test:test")
```

Comment:
I'm not sure what the RFC says, but are these allowed?
  * `test:` (empty password, maybe valid)
  * `:test` (empty username, probably invalid)
  * `:` (both empty - is it then a username by default, with empty password? or the other way around)

We should make sure to cover these cases: in the tests as well as the cli-parsing part.